src/js/containers/awardV2/shared/FederalAccountsSummaryContainer.jsx

```diff
@@ -0,0 +1,94 @@
+/**
+  * FederalAccountsSummaryContainer.jsx
+  * Created by Lizzie Salita 8/16/19
+  **/
+
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { isCancel } from 'axios';
+
+import { fetchIdvFundingSummary } from 'helpers/idvHelper';
+import { fetchAwardFundingSummary } from 'helpers/awardSummaryHelper';
+import BaseFundingRollup from 'models/v2/awardsV2/BaseFundingRollup';
+import FederalAccountsSummary from 'components/awardv2/shared/federalAccounts/FederalAccountsSummary';
+
+const propTypes = {
+    awardId: PropTypes.string,
+    category: PropTypes.string,
+    jumpToFederalAccountsHistory: PropTypes.func
+};
+
+export class FederalAccountsSummaryContainer extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            inFlight: true,
+            summary: null
+        };
+
+        this.request = null;
+    }
+
+    async componentDidMount() {
+        if (this.props.awardId && this.props.category) {
+            await this.getAwardMetaData();
+        }
+    }
+
+    async componentDidUpdate(prevProps) {
+        if (prevProps.awardId !== this.props.awardId && (this.props.awardId && this.props.category)) {
+            await this.getAwardMetaData();
+        }
+    }
+
+    async getAwardMetaData() {
+        if (this.request) {
+            this.request.cancel();
+        }
+        if (this.props.category === 'idv') {
```

Comment:
FYSA I have a PR for the download button where I'm introducing a pattern where we abstract the condition into a new helper that takes the awardid and category as params and returns either the correct endpoint:

- [here's the helper that abstracts the conditional logic](https://github.com/fedspendingtransparency/usaspending-website/pull/1238/files?file-filters%5B%5D=.js#diff-2253e63fec011dd0d2986a9aa60409fbR113)
- [here's where its implemented](https://github.com/fedspendingtransparency/usaspending-website/pull/1238/files#diff-5d272f09dbb41cf6b50684bce394a07aR149)

 Bringing this to your attention but I think if we don't use this pattern here I should probably not use it in my pr for consistency sake?

Reply:
Nice! I think that makes sense there, but here the API calls are in separate files (`idvHelper` and `awardSummaryHelper`) so I think it's ok if they're inconsistent.